fw/http.c

```diff
@@ -5906,6 +5907,21 @@ tfw_http_req_process(TfwConn *conn, TfwStream *stream, struct sk_buff *skb)
 					     " processing error");
 		TFW_INC_STAT_BH(clnt.msgs_otherr);
 	}
+
+	if (unlikely(test_bit(TFW_HTTP_B_CONN_UPGRADE, req->flags)
+		     && test_bit(TFW_HTTP_B_UPGRADE_WEBSOCKET, req->flags)))
+	{
+		/*
+		 * We take an additional refcnt one to prevent access to the
+		 * freed connection when racing between client and server
+		 * tfw_ws_conn_drop()
+		 */
+		if (!conn->get_refcnt) {
+			tfw_connection_get(conn);
+			conn->get_refcnt = true;
```

Comment:
My comment not about current section. However, pipelining of "upgrade" requests are leads to ws connection "object" leaks. I believe we should reject all upgrade requests after first successful upgrade. You can to reproduce it using this [test](https://github.com/tempesta-tech/tempesta-test/blob/kt-1647-poc/ws/test_ws_ping.py#L473) from my branch.

Reply:
What does it mean (reject)? Send a specific response?